tests/integration/test_writes/test_partitioned_writes.py

```diff
@@ -1126,3 +1127,25 @@ def test_append_multiple_partitions(
         """
     )
     assert files_df.count() == 6
+
+
+@pytest.mark.integration
+def test_pyarrow_overflow(session_catalog: Catalog) -> None:
+    """Test what happens when the offset is beyond 32 bits"""
+    identifier = "default.arrow_table_overflow"
+    try:
+        session_catalog.drop_table(identifier=identifier)
+    except NoSuchTableError:
+        pass
+
+    x = pa.array([random.randint(0, 999) for _ in range(30_000)])
```

Comment:
i wonder if random here is making the test non-deterministic. since we're testing the overflow, what if we just use `999` or do `random.randint(900, 999)`

Reply:
I would expect the overflow to come from the `StringType` column, but I agree to make this test deterministic 👍

Reply:
```suggestion
    x = pa.array([1925 for _ in range(30_000)])
```